src/openrct2/interface/Window.cpp

```diff
@@ -2167,3 +2167,18 @@ void WindowBase::ResizeFrameWithPage()
     widgets[3].right = width - 1;
     widgets[3].bottom = height - 1;
 }
+
+void WindowBase::ResizeDropdown(WidgetIndex widgetIndex, const ScreenCoordsXY& origin, const ScreenSize& size)
```

Comment:
This is indeed very close to what I had in mind, good job! :+1: 

I’d like you to change one thing: I think it is better to only pass the width and not the height. This is because I would to make dropdown bigger on touch interfaces (which will be a future PR), which in turns means their height cannot be hardcoded by the window. There should be some kind of constant somewhere that you can use. Otherwise, you could simply create one.

Reply:
Do you know where to look for one? The window I cribbed the positioning code used a hardcoded number with no named constant

Reply:
I just added a new `DROPDOWN_HEIGHT` constant to `src/openrct2/interface/Window.h` with all the others.